azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -403,6 +416,30 @@ def on_method_request_received(self):
     def on_twin_desired_properties_patch_received(self):
         pass
 
+    @property
+    def on_connection_state_change(self):
+        return self._handler_manager.on_connection_state_change
+
+    @on_connection_state_change.setter
+    def on_connection_state_change(self, value):
+        self._handler_manager.on_connection_state_change = value
+
+    # @property
+    # def on_new_sastoken_required(self):
+    #     return self._handler_manager.on_new_sastoken_required
```

Comment:
Just curious, what's the plan for these handlers?

Reply:
Add them in the next PR(s). I built some of their infrastructure in this PR because I was testing faked versions of them at client level to ensure that the infrastructure could handle these new "client event" handlers working at the same time.

In order to implement these, I'd just uncomment them at client level (where they're already functional), and then just hook them up to a trigger that comes out of the pipeline when the event occurs.